DataStructures/Lists/CircularBuffer.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+
+namespace DataStructures.Lists {
+    public class CircularBuffer<T> {
+        /// <summary>
+        /// Initializes a circular buffer with initial length of 10
+        /// </summary>
+        public CircularBuffer(bool canOverride=true) : this(10,canOverride) {
+        }
+
+        /// <summary>
+        /// Initializes a circular buffer with given length
+        /// </summary>
+        /// <param name="length">The length of the buffer</param>
+        public CircularBuffer(int length, bool canOverride=true) {
+            if (length < 1) {
+                throw new ArgumentOutOfRangeException("length can not be zero or negative");
+            }
+            _circularBuffer = new T[length + 1];
+            _end = 0;
+            _start = 0;
+            CanOverride = canOverride;
+        }
+
+        /// <summary>
+        /// Writes value to the back of the buffer
+        /// </summary>
+        /// <param name="value">value to be added to the buffer</param>
+        public void Write(T value) {
+            if (CanOverride) {
+                InsertData(value);
+            }
+            else {
+                DontOverrides(value);
+            }
+        }
+
+        /// <summary>
+        /// Inserts data into the buffer when it is not filled up
+        /// </summary>
+        /// <param name="value"></param>
+        private void DontOverrides(T value) {
+            if (IsFilledUp) {
+                return;
```

Comment:
Maybe we should throw exception (ArgumentOutOfRange  in this case? But here i am not sure, so open for discussion.

From use-case perspective if i add some property i expect it'll be added or i get some information back (if it won't be added), in this case it can be exception.

Reply:
Throwing an exception will make more sense than just leaving the user with no clue as to what happened to the data he tried to insert. The custom exception name is excellent. I've implemented it.

Reply:
I also added test cases to confirm that the exception is throw appropriately.